zerver/lib/markdown/__init__.py

```diff
@@ -1719,10 +1720,27 @@ def clean(self, line: str) -> str:
         line = mention.MENTIONS_RE.sub(lambda m: "@_**{}**".format(m.group("match")), line)
         # Silence all the user group mentions inside blockquotes
         line = mention.USER_GROUP_MENTIONS_RE.sub(lambda m: "@_*{}*".format(m.group("match")), line)
+        # Don't convert emoticons present inside blockquotes.
+        # When a user, who has `Convert emoticons before sending` enabled, quotes the message
+        # containing emoticon, sent by another user, who has the same setting disabled, creates
+        # a confussion for the latter user as they see those as the converted emoticon.
+        line = self.COMPILED_EMOTICON_RE.sub(
+            lambda m: self.escape_emoticon(m.group("emoticon")), line
+        )
```

Comment:
I feel like we could do this better by looking at the state of the parse tree, and just treating the "emoticons" data set as though the feature is disabled?

Reply:
Do you mean I should just set `translate_emoticons = False` when we detect that emoticons are present inside the blockquote?
Doing that would not let the emoticons present **outside** the blockquote to convert into an emoji.

Reply:
No, I mean something else.  I don't have a concrete plan, since it depends on the order of processing, but if we've already processed the blockquote logic at the time this runs, we can inspect our parents in the parse tree to see if any are blockquote objects, and if so, include that case in the `return None` conditional below.

```
    def handleMatch(self, match: Match[str]) -> Optional[Element]:                                   
        db_data = self.md.zulip_db_data                                                              
        if db_data is None or not db_data["translate_emoticons"]:                                    
            return None                                                                              
                                                                                                     
        emoticon = match.group("emoticon")                                                           
        translated = translate_emoticons(emoticon)                                                   
        name = translated[1:-1]                                                                      
        return make_emoji(name_to_codepoint[name], translated)                                       
```